app/javascript/hyacinth_ui_v1/components/controlled_vocabularies/ControlledVocabularyEdit.js

```diff
@@ -1,23 +1,34 @@
 import React from 'react';
+import { useParams } from 'react-router-dom';
+import { useQuery } from '@apollo/react-hooks';
 
 import ContextualNavbar from '../layout/ContextualNavbar';
 import ControlledVocabularyForm from './ControlledVocabularyForm';
+import { getVocabularyQuery } from '../../graphql/vocabularies';
+import GraphQLErrors from '../ui/GraphQLErrors';
 
-class ControlledVocabularyEdit extends React.PureComponent {
-  render() {
-    const { match: { params: { stringKey } } } = this.props;
+function ControlledVocabularyEdit() {
+  const { stringKey } = useParams();
 
-    return (
-      <div className="m-3">
-        <ContextualNavbar
-          title="Update Controlled Vocabulary"
-          rightHandLinks={[{ link: '/controlled_vocabularies', label: 'Back to Controlled Vocabulary' }]}
-        />
+  const { loading, error, data } = useQuery(
+    getVocabularyQuery, {
+      variables: { stringKey },
+    },
+  );
 
-        <ControlledVocabularyForm formType="edit" stringKey={stringKey} key={stringKey} />
-      </div>
-    );
-  }
+  if (loading) return (<></>);
+  if (error) return (<GraphQLErrors errors={error} />);
+
+  return (
+    <div className="m-3">
+      <ContextualNavbar
+        title="Update Controlled Vocabulary"
+        rightHandLinks={[{ link: '/controlled_vocabularies', label: 'Back to Controlled Vocabulary' }]}
+      />
+
+      <ControlledVocabularyForm formType="edit" vocabulary={data.vocabulary} key={stringKey} />
```

Comment:
It's not immediately obvious to me why `key` is necessary here.  Anything do with preventing component caching?

Reply:
if that component is used in contexts where the supporting data is refetched?

Reply:
I thought that a different value of `vocabulary={data.vocabulary}` each time would cause it to show a different vocabulary, and the key wouldn't be necessary.  I'll ticket as something to look into later?

Reply:
After our discussion, I can go head and remove the key if that's what you'd prefer.

Reply:
That would be great, thanks!

Reply:
Removed key.